models/shared.py

```diff
@@ -34,10 +34,10 @@ def find_set(self, query):
             if str(query).lower() in [n.lower() for n in [kfset["SetName"], kfset["ShortName"]]] or str(query).lower() == str(kfset["SetNumber"]).lower():
                 return kfset
     def assigned_set_name(self, set_query, card_num=None):
-        set_num = self.find_set(set_query)["SetNumber"]
-        if str(set_num) == "453":
+        if str(set_query) == "453":
```

Comment:
I'm a bit confused about this change - did it not work the other way? The set_query param is meant to be either a number or the short or long name of the set, so whichever you have you can look up data about it.

I don't think this function is used much so it's not a big deal.

Reply:
It was crashing. I can look again at what was going on here.

Reply:
Okay, I went back to it and remember what was up. 453 is the anomaly set, which isn't in this dataset to search:

(Pdb) print(self.sets.keys())
dict_keys(['341', '435', '452', '479', '496', '600', '609', '700', '722', '737', '800', '855', '874', '892', '999', 'KFA001', 'KFA002'])

So the end of this line

`set_num = self.find_set(set_query)["SetNumber"]`

crashes because the object is None and there is no "SetNumber" key to index.

This diff for some reason isn't showing that I moved the find_set() call just as is right below the anomaly part. It looked to me like we have to deal with anomalies, but then do exactly the find_set() approach:

```
        set_num = self.find_set(set_query)["SetNumber"] # <-- existing code
        if str(set_num) == "453":
        #if str(set_query) == "453":    <--- or handle anomaly specially, first?
            icard_num = int(card_num[1:])
            return self.anomaly_sets[icard_num]["SetName"]
        set_num = self.find_set(set_query)["SetNumber"]  # <-- then do find_set()?
        return self.sets[str(set_num)]["SetName"]
```